File: 645-SetMismatch/645-SetMismatch.cpp
// Last updated: 13/07/2026, 22:17:39
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        const int n = nums.size();
        vector<bool> found(n, false);
        vector<int> result(2, 0);

        for (auto& val : nums){
            if (found[val - 1]) result[0] = val;
            else found[val - 1] = true;
        }
        for (int i = 0; i < n; ++i) if (!found[i]) result[1] = i + 1;
        return result;
    }
};

File: 645-SetMismatch/645-SetMismatch.py
# Last updated: 13/07/2026, 22:17:46
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        set_list = set()
        duplicate = -1
        for num in nums:
            if (num in set_list):
                duplicate = num
            set_list.add(num)

        length = len(nums)

        intended_target = length * (length + 1) // 2
        actual_sum = sum(set_list)

        removed = intended_target - actual_sum

        return [duplicate , removed]